manifests/config.pp

```diff
@@ -185,10 +186,14 @@
         include 'systemd'
         systemd::unit_file {'prometheus.service':
           content => template('prometheus/prometheus.systemd.erb'),
+          notify  => Class['prometheus::run_service'],
         }
+        -> Class['systemd::systemctl::daemon_reload']
+        -> Class['prometheus::run_service']
```

Comment:
This looks redundant as you've already got a `notify` metaparameter.  Probably best not to pick the two styles either.  ie `-> Class['systemd::systemctl::daemon_reload']` could be `before => Class['systemd::systemctl::daemon_reload']` instead.

Reply:
It did not work. I don't what else we can do to improve this.

Reply:
I am able to make a small change but still not able to remove notify.

Reply:
Sorry, it wasn't the `notify =>` I thought could go.  Only that if you already had a `notify` then an additional resource relationship via the `->` syntax wouldn't add anything extra. I hit this bug today too.  I'll have a closer look on Monday.  Perhaps there's another way where reloading systemd isn't even needed. (eg. by moving the flag options to an environment variable or something).